<a href="https://www.kaggle.com/code/elizabethgels/cyclistic-case-study-sql-and-tableau?scriptVersionId=139552539" target="_blank"><img align="left" alt="Kaggle" title="Open in Kaggle" src="https://kaggle.com/static/images/open-in-kaggle.svg"></a>

# CYCLISTIC CASE STUDY: HOW DOES A BIKE-SHARE NAVIGATE SPEEDY SUCCESS?

1. [ASK](#1)
    1. [ABOUT THE COMPANY](#2)
    1. [CHARACTERS AND TEAMS](#3)
    1. [BUSINESS TASK](#4)
1. [PREPARE](#5)
    1. [DATA SOURCE](#6)
    1. [ANALYTICAL TOOL](#7)
1. [PROCESS](#8)
    1. [DATA CLEANING](#9)
1. [ANALYZE](#10)
1. [SHARE](#11)
1. [ACT](#12)
    1. [CONCLUSIONS](#13)
    1. [RECOMMENDATIONS](#14)


<a id="1"></a>
# 1. ASK


<a id="2"></a>
### A. ABOUT THE COMPANY


In 2016, Cyclistic launched a successful bike-share offering. Since then, the program has grown to a fleet of 5,824 bicycles that are geotracked and locked into a network of 692 stations across Chicago. The bikes can be unlocked from one station and returned to any other station in the system anytime.

Until now, Cyclistic's marketing strategy relied on building general awareness and appealing to broad consumer segments. One approach that helped make these things possible was the flexibility of its pricing plans: single-ride passes, full-day passes, and annual memberships. Customers who purchase single-ride or full-day passes are referred to as Casual riders. Customers who purchase annual memberships are Cyclistic Members.

Cyclistic's finance analysts have concluded that annual Members are much more profitable than Casual riders. Although the pricing flexibility helps Cyclistic attract more customers, the director of marketing (Lily Moreno) believes that maximizing the number of annual Members will be key to future growth. Rather than creating a marketing campaign that targets all-new customers, Moreno believes there is a very good chance to convert Casual riders into Members. She notes that Casual riders are already aware of the Cyclistic program and have chosen Cyclistic for their mobility needs. Moreno has set a clear goal: Design marketing strategies aimed at converting Casual riders into annual Members. In order to do that, however, the marketing analyst team needs to better understand how annual Members and Casual riders differ, why Casual riders would buy a membership, and how digital media could affect their marketing tactics. 

<a id="3"></a>
### B. CHARACTERS AND TEAMS

* Cyclistic: A bike-share program that features more than 5,800 bicycles and 600 docking stations. Cyclistic sets itself apart by also offering reclining bikes, hand tricycles, and cargo bikes, making bike-share more inclusive to people with disabilities and riders who can't use a standard two-wheeled bike.
* Lily Moreno: The director of marketing, Moreno is responsible for the development of campaigns and initiatives to promote the bike-share program. These may include email, social media, and other channels.
* Cyclistic marketing analytics team: A team of data analysts who are responsible for collecting, analyzing, and reporting data that helps guide Cyclistic marketing strategy.
* Cyclistic executive team: The notoriously detail-oriented executive team will decide whether to approve the recommended marketing program.

<a id="4"></a>
### C. BUSINESS TASK

Working as a data analyst in the marketing analytics team, I will analyze the Cyclistic historical bike trip data to gain insight into how Casual riders and annual Members use Cyclistic bikes differently. From these insights, the marketing team will design a new strategy to convert Casual riders into annual Members. But first, Cyclistic executives must approve the recommendations, so they must be backed up with compelling data insights and visualizations.



<a id="5"></a>
# 2. PREPARE


<a id="6"></a>
### A. DATA SOURCE

The data is public, and it has been made available by Motivate International Inc. under this license [license](https://ride.divvybikes.com/data-license-agreement). The datasets can be downloaded [here](https://divvy-tripdata.s3.amazonaws.com/index.html). There is available data from 2018 but I will use the last twelve months of Cyclistic's historical trip data.; from March 2022 through February 2023.

Note: Although Cyclistic is a fictional company, the datasets are appropriate to answer the business questions.

<a id="7"></a>
### B. ANALYTICAL TOOL

The tool chosen for this analysis is SQL, programmed in BigQuery (Google Cloud). For the visualizations I'll use Tableau.

In the code below, we connect to BigQuery to pull the dataset' tables that I will use in the analysis.

In [1]:
# Identify the project ID in BQ
PROJECT_ID = 'my-project-010823-374220'

# Import the BQ API Client library
from google.cloud import bigquery
client = bigquery.Client(project=PROJECT_ID, location='US')

# Construct a reference to the BigQuery dataset that is within the project
dataset_ref = client.dataset('cyclistic_data', project=PROJECT_ID)

# Make an API request to fetch the dataset
dataset = client.get_dataset(dataset_ref)

# Make a list of all the tables in the dataset
tables = list(client.list_tables(dataset))

# Print names of all tables in the dataset
for table in tables:  
    print(table.table_id)


202203_tripdata
202204_tripdata
202205_tripdata
202206_tripdata
202207_tripdata
202208_tripdata
202209_tripdata
202210_tripdata
202211_tripdata
202212_tripdata
202301_tripdata
202302_tripdata
Unique_Coordinates
combined_tripdata
combined_tripdatav2
combined_tripdatav3


* 202203_tripdata.csv
* 202204_tripdata.csv
* 202205_tripdata.csv
* 202206_tripdata.csv
* 202207_tripdata.csv
* 202208_tripdata.csv
* 202209_tripdata.csv
* 202210_tripdata.csv
* 202211_tripdata.csv
* 202212_tripdata.csv
* 202301_tripdata.csv
* 202302_tripdata.csv

Next, we recognize that these twelve tables have the same schema. Therefore we will use the UNION statement to put them all together in one new table called "combined_tripdata".

In [2]:
import warnings
warnings.filterwarnings('ignore')

query1 = """

CREATE TABLE my-project-010823-374220.cyclistic_data.combined_tripdata AS

  (SELECT * FROM `my-project-010823-374220.cyclistic_data.202203_tripdata` 
    UNION ALL
  SELECT * FROM `my-project-010823-374220.cyclistic_data.202204_tripdata` 
    UNION ALL
  SELECT * FROM `my-project-010823-374220.cyclistic_data.202205_tripdata` 
    UNION ALL
  SELECT * FROM `my-project-010823-374220.cyclistic_data.202206_tripdata` 
    UNION ALL
  SELECT * FROM `my-project-010823-374220.cyclistic_data.202207_tripdata` 
    UNION ALL
  SELECT * FROM `my-project-010823-374220.cyclistic_data.202208_tripdata` 
    UNION ALL
  SELECT * FROM `my-project-010823-374220.cyclistic_data.202209_tripdata` 
    UNION ALL
  SELECT * FROM `my-project-010823-374220.cyclistic_data.202210_tripdata` 
    UNION ALL
  SELECT * FROM `my-project-010823-374220.cyclistic_data.202211_tripdata` 
    UNION ALL
  SELECT * FROM `my-project-010823-374220.cyclistic_data.202212_tripdata` 
    UNION ALL
  SELECT * FROM `my-project-010823-374220.cyclistic_data.202301_tripdata` 
    UNION ALL
  SELECT * FROM `my-project-010823-374220.cyclistic_data.202302_tripdata`); 
   """
# Construct a reference to the "combined_tripdata" table
table_ref = dataset_ref.table("combined_tripdata")

# API request to fetch the table
table = client.get_table(table_ref)

# Preview the first ten lines of the table
client.list_rows(table, max_results=10).to_dataframe()

ride_id  rideable_type                started_at  \
0  20E71EE7F849E218  electric_bike 2022-07-04 08:15:24+00:00   
1  9FA1B60F33BE49B4  electric_bike 2022-10-20 16:34:04+00:00   
2  710B8497D8DFA76F  electric_bike 2022-10-25 00:39:07+00:00   
3  1199AFD37CF52C2F  electric_bike 2022-09-03 13:17:17+00:00   
4  52D84FC0602FF2C9  electric_bike 2022-10-30 13:42:15+00:00   
5  8C311E758831B6B6  electric_bike 2022-09-17 20:35:56+00:00   
6  A1139EBA731B8699  electric_bike 2022-07-03 22:55:12+00:00   
7  C5FC6E0D8260BD87  electric_bike 2022-09-16 05:10:35+00:00   
8  D86883BE2B8867C3  electric_bike 2022-08-30 20:57:14+00:00   
9  2F78D9024FC6C1D0  electric_bike 2022-09-10 17:05:13+00:00   

                   ended_at start_station_name start_station_id  \
0 2022-07-04 08:48:19+00:00               None             None   
1 2022-10-20 16:39:43+00:00               None             None   
2 2022-10-25 01:13:26+00:00               None             None   
3 2022-09-03 13:28:02+00:00               None             None   
4 2022-10-30 13:52:19+00:00               None             None   
5 2022-09-17 20:57:33+00:00               None             None   
6 2022-07-03 23:09:54+00:00               None             None   
7 2022-09-16 05:20:11+00:00               None             None   
8 2022-08-30 21:03:31+00:00               None             None   
9 2022-09-10 17:40:57+00:00               None             None   

  end_station_name end_station_id  start_lat  start_lng  end_lat  end_lng  \
0             None           None      42.00     -87.71     42.0   -87.70   
1             None           None      42.02     -87.67     42.0   -87.66   
2             None           None      42.00     -87.66     42.0   -87.66   
3             None           None      42.01     -87.68     42.0   -87.66   
4             None           None      41.98     -87.69     42.0   -87.70   
5             None           None      42.03     -87.70     42.0   -87.70   
6             None           None      41.97     -87.67     42.0   -87.66   
7             None           None      42.01     -87.67     42.0   -87.68   
8             None           None      42.00     -87.71     42.0   -87.70   
9             None           None      42.00     -87.67     42.0   -87.67   

  member_casual  
0        member  
1        member  
2        casual  
3        casual  
4        casual  
5        member  
6        casual  
7        member  
8        casual  
9        member

In [3]:
table.schema

[SchemaField('ride_id', 'STRING', 'NULLABLE', None, (), None),
 SchemaField('rideable_type', 'STRING', 'NULLABLE', None, (), None),
 SchemaField('started_at', 'TIMESTAMP', 'NULLABLE', None, (), None),
 SchemaField('ended_at', 'TIMESTAMP', 'NULLABLE', None, (), None),
 SchemaField('start_station_name', 'STRING', 'NULLABLE', None, (), None),
 SchemaField('start_station_id', 'STRING', 'NULLABLE', None, (), None),
 SchemaField('end_station_name', 'STRING', 'NULLABLE', None, (), None),
 SchemaField('end_station_id', 'STRING', 'NULLABLE', None, (), None),
 SchemaField('start_lat', 'FLOAT', 'NULLABLE', None, (), None),
 SchemaField('start_lng', 'FLOAT', 'NULLABLE', None, (), None),
 SchemaField('end_lat', 'FLOAT', 'NULLABLE', None, (), None),
 SchemaField('end_lng', 'FLOAT', 'NULLABLE', None, (), None),
 SchemaField('member_casual', 'STRING', 'NULLABLE', None, (), None)]

The "combined_tripdata" table has 5'829,084 rows. It contains all rides that occured during March 2022 through February 2023. Each ride is identified by a unique ride_id, it includes the time when the ride started (started_at), the time when it ended (ended_at), the starting station (start_station_name and start_station_id), the ending station (end_station_name and end_station_id) and the coordinates for both, the start and ending stations (start_lat, start_lng, end_lat, end_lng). The data is grouped by Casual and annual Members (member_casual) and the type of ride (rideable_type) which possible values are electric bike or classic bike.

The fields in the table have the correct data types, except that we will change start_at and ended_at to DATETIME later in the process phase.

<a id="8"></a>
# 3. PROCESS


<a id="9"></a>
### A. DATA CLEANING

The following NULL values were found:

* start_station_name has 850,418 NULLS 
* end_station_name has 909,038 NULLS  
* start_station_id 850,550 has NULLS
* end_station_id 909,179 has NULLS 
* end_lat and end_lng have 5,938 NULLS 

We decided to keep all null values in the table because ride id, started_at and ended_at have no nulls. Therefore we assume that these observations are valid trips that need to be counted.

In [4]:
query2="""

SELECT * 
FROM my-project-010823-374220.cyclistic_data.combined_tripdata
WHERE start_station_name IS NULL AND end_station_name IS NULL AND
start_station_id IS NULL AND end_station_id IS NULL;
"""
query_job = client.query(query2)

# API request - run the query, and convert the results to a pandas DataFrame
existing_nulls = query_job.to_dataframe()

# Print the first five rows of the DataFrame
existing_nulls.head(5)

ride_id  rideable_type                started_at  \
0  5A5E78803481ABCF  electric_bike 2022-09-28 10:28:13+00:00   
1  FD9202270D0ED76A  electric_bike 2022-08-31 12:46:17+00:00   
2  379525FADB59A9C5  electric_bike 2022-03-21 16:20:05+00:00   
3  727C4AE9C08ED981  electric_bike 2022-08-02 08:10:56+00:00   
4  B869AA78E561F9AA  electric_bike 2022-04-16 18:31:19+00:00   

                   ended_at start_station_name start_station_id  \
0 2022-09-28 10:47:24+00:00               None             None   
1 2022-08-31 13:35:00+00:00               None             None   
2 2022-03-21 16:48:49+00:00               None             None   
3 2022-08-02 08:36:53+00:00               None             None   
4 2022-04-16 18:45:03+00:00               None             None   

  end_station_name end_station_id  start_lat  start_lng  end_lat  end_lng  \
0             None           None      41.95     -87.83    42.00   -87.83   
1             None           None      41.75     -87.57    41.75   -87.57   
2             None           None      41.74     -87.55    41.75   -87.61   
3             None           None      41.75     -87.57    41.75   -87.57   
4             None           None      41.76     -87.57    41.75   -87.59   

  member_casual  
0        member  
1        casual  
2        member  
3        casual  
4        casual

In [5]:
query3="""

SELECT *
FROM my-project-010823-374220.cyclistic_data.combined_tripdata
WHERE end_lng IS NULL AND end_lng IS NULL;
"""
query_job = client.query(query3)

coord_nulls = query_job.to_dataframe()

coord_nulls.head(5)

ride_id rideable_type                started_at  \
0  A97ACE2D78CDFAB7  classic_bike 2022-11-26 18:37:31+00:00   
1  7A7511BA12A21290  classic_bike 2022-07-26 10:39:42+00:00   
2  89B2923C17A9500A          None 2022-10-07 15:07:57+00:00   
3  D1289A34B8E55A0B  classic_bike 2022-06-27 19:08:54+00:00   
4  DD7AD6B8D0A4B002  classic_bike 2022-07-07 19:00:33+00:00   

                   ended_at         start_station_name start_station_id  \
0 2022-11-27 19:37:25+00:00     Paulina St & Howard St              515   
1 2022-07-27 11:39:35+00:00  Conservatory Dr & Lake St              518   
2 2022-10-13 13:25:00+00:00   Glenwood Ave & Touhy Ave              525   
3 2022-06-28 20:08:50+00:00    Laramie Ave & Kinzie St              530   
4 2022-07-08 20:00:27+00:00    Kenton Ave & Madison St              537   

  end_station_name end_station_id  start_lat  start_lng  end_lat  end_lng  \
0             None           None  42.019159 -87.673573      NaN      NaN   
1             None           None  41.885502 -87.716866      NaN      NaN   
2             None           None  42.012701 -87.666058      NaN      NaN   
3             None           None  41.887832 -87.755527      NaN      NaN   
4             None           None  41.880708 -87.741018      NaN      NaN   

  member_casual  
0        casual  
1        casual  
2        casual  
3        casual  
4        casual

No duplicates were found when querying ride_ids.

In [6]:
query4="""

SELECT 
  ride_id, COUNT(ride_id) AS rides
FROM my-project-010823-374220.cyclistic_data.combined_tripdata
GROUP BY ride_id
HAVING COUNT(ride_id)>1;
"""
query_job = client.query(query4)

duplicates = query_job.to_dataframe()

duplicates.head()


Empty DataFrame
Columns: [ride_id, rides]
Index: []

Inspecting the minimun and maximun values for started_at and ended_at. 

started_at:

* Min= 2022-03-01 00:00:19 UTC
* Max = 2023-02-28 23:59:31 UTC

ended_at:

* Min= 2022-03-01 00:04:30 UTC
* Max = 2023-03-06 15:09:53 UTC

In [7]:
query5="""

SELECT
  MIN(started_at) AS min_started_at,
  MAX(started_at) AS max_started_at
FROM my-project-010823-374220.cyclistic_data.combined_tripdata;
"""
query_job = client.query(query5)

started_dates = query_job.to_dataframe()

started_dates.head()


min_started_at            max_started_at
0 2022-03-01 00:00:19+00:00 2023-02-28 23:59:31+00:00

In [8]:
query6="""

SELECT
  MIN(ended_at) AS min_ended_at,
  MAX(ended_at) AS max_ended_at
FROM my-project-010823-374220.cyclistic_data.combined_tripdata;
"""
query_job = client.query(query6)

ended_dates = query_job.to_dataframe()

ended_dates.head()


min_ended_at              max_ended_at
0 2022-03-01 00:04:30+00:00 2023-03-06 15:09:53+00:00

The values found for rideable_type are: Electric_bike, classic_bike and docked_bike.

We assume that all bikes are locked into a station so all bikes are docked bikes. Therefore the value "docked bike" seems to be a mistake, and we're replacing it by NULLs. 

In [9]:
import warnings
warnings.filterwarnings('ignore')

query7="""

SELECT
  rideable_type,
  COUNT (DISTINCT ride_id) AS number_of_trips
FROM my-project-010823-374220.cyclistic_data.combined_tripdata
GROUP BY rideable_type
ORDER BY number_of_trips DESC;

/* UPDATE my-project-010823-374220.cyclistic_data.combined_tripdata 
SET rideable_type=NULL WHERE rideable_type="docked_bike"; */
"""
query_job = client.query(query7)

rideable_count = query_job.to_dataframe()

rideable_count.head()

rideable_type  number_of_trips
0  electric_bike          2983084
1   classic_bike          2666915
2           None           179085

We create a new table to calculate new fields: Day, weekday_name, month_name, hour_of_day, ride_length_min, dist_meters and dist_miles.

 Also, we're changing started_at and ended_at to DATETIME data type.

In [10]:
import warnings
warnings.filterwarnings('ignore')

query8="""

CREATE TABLE my-project-010823-374220.cyclistic_data.combined_tripdatav2 AS
SELECT 
  ride_id, rideable_type, start_station_name, start_station_id, end_station_name,
  end_station_id, member_casual,start_lat, start_lng, end_lat, end_lng,
  DATE(started_at) AS day,
  FORMAT_DATE('%A', DATE(started_at)) AS weekday_name,
  FORMAT_DATE('%B', DATE(started_at)) AS month_name,
  EXTRACT(HOUR FROM (CAST(started_at AS DATETIME))) AS hour_of_day,
  CAST(ended_at AS DATETIME) AS ended_at_dt,
  CAST(started_at AS DATETIME) AS started_at_dt,
  (DATE_DIFF(CAST(ended_at AS DATETIME), CAST(started_at AS DATETIME), second))/60 AS ride_length_min,
  ST_Distance(ST_GeogPoint(start_lng, start_lat), ST_GeogPoint(end_lng, end_lat)) AS dist_meters,
  (ST_Distance(ST_GeogPoint(start_lng, start_lat), ST_GeogPoint(end_lng, end_lat)))/1609.344 AS dist_miles
FROM my-project-010823-374220.cyclistic_data.combined_tripdata;
"""
table_ref = dataset_ref.table("combined_tripdatav2")

table = client.get_table(table_ref)

client.list_rows(table, max_results=10).to_dataframe()


ride_id  rideable_type start_station_name start_station_id  \
0  AD200CA94DB59BE3  electric_bike               None             None   
1  44202C98B1DE5B7C  electric_bike               None             None   
2  D5244A734E83BD15  electric_bike               None             None   
3  E1FD7C094F2AFFAD  electric_bike               None             None   
4  869FB6F403263FC3  electric_bike               None             None   
5  458C7DC447896739  electric_bike               None             None   
6  E96A2C57C85FCFD6  electric_bike               None             None   
7  D8DF16E0DF0B9B44  electric_bike               None             None   
8  9CD15AF3622BA19F  electric_bike               None             None   
9  3D2C77D878453566  electric_bike               None             None   

                        end_station_name end_station_id member_casual  \
0                                   None           None        casual   
1                                   None           None        casual   
2                                   None           None        member   
3                                   None           None        member   
4                                   None           None        member   
5                                   None           None        casual   
6                                   None           None        casual   
7                                   None           None        member   
8  Public Rack - Western Ave & Devon Ave            931        member   
9                                   None           None        member   

   start_lat  start_lng  end_lat  end_lng         day weekday_name month_name  \
0      42.01     -87.68     42.0   -87.67  2022-11-11       Friday   November   
1      41.99     -87.66     42.0   -87.66  2022-07-17       Sunday       July   
2      42.06     -87.72     42.0   -87.68  2022-07-04       Monday       July   
3      42.00     -87.71     42.0   -87.71  2022-07-10       Sunday       July   
4      42.00     -87.66     42.0   -87.66  2022-07-06    Wednesday       July   
5      42.01     -87.71     42.0   -87.71  2022-07-23     Saturday       July   
6      41.98     -87.71     42.0   -87.69  2022-04-24       Sunday      April   
7      41.99     -87.67     42.0   -87.70  2022-03-21       Monday      March   
8      42.00     -87.66     42.0   -87.69  2022-06-10       Friday       June   
9      41.98     -87.66     42.0   -87.67  2022-09-23       Friday  September   

   hour_of_day         ended_at_dt       started_at_dt  ride_length_min  \
0           14 2022-11-11 14:58:11 2022-11-11 14:52:44         5.450000   
1           17 2022-07-17 18:50:05 2022-07-17 17:53:17        56.800000   
2           12 2022-07-04 13:02:55 2022-07-04 12:28:23        34.533333   
3           13 2022-07-10 14:14:51 2022-07-10 13:44:05        30.766667   
4           18 2022-07-06 18:18:38 2022-07-06 18:16:42         1.933333   
5           22 2022-07-23 22:09:44 2022-07-23 22:03:07         6.616667   
6           15 2022-04-24 15:43:52 2022-04-24 15:33:29        10.383333   
7            0 2022-03-21 00:15:46 2022-03-21 00:03:16        12.500000   
8           22 2022-06-10 22:28:38 2022-06-10 22:19:23         9.250000   
9           13 2022-09-23 13:22:41 2022-09-23 13:13:47         8.900000   

   dist_meters  dist_miles  
0  1385.339883    0.860810  
1  1111.951012    0.690934  
2  7444.916237    4.626056  
3     0.000000    0.000000  
4     0.000000    0.000000  
5  1111.951012    0.690934  
6  2770.912119    1.721765  
7  2717.158514    1.688364  
8  2479.021909    1.540393  
9  2372.507883    1.474208

We found a problem with ride_length (calculated in minutes). The maximun value is 29 days and the minimun is a negative number.

In [11]:
query9="""

SELECT
  AVG(ride_length_min) AS mean_ride_length,
  MAX(ride_length_min) AS max_ride_length,
  MIN(ride_length_min) AS min_ride_length
FROM my-project-010823-374220.cyclistic_data.combined_tripdatav2;
"""
query_job = client.query(query9)

rideable_count = query_job.to_dataframe()

rideable_count.head()

mean_ride_length  max_ride_length  min_ride_length
0         19.217434         41387.25        -10353.35

We filter any ride length that is negative, and over 7 days. We assume that after 7 days the bike is considered lost. 

We filter 1,054 observations.

In [12]:
query10="""
CREATE TABLE my-project-010823-374220.cyclistic_data.combined_tripdatav3 AS
SELECT *
FROM my-project-010823-374220.cyclistic_data.combined_tripdatav2
WHERE ride_length_min >0 AND ride_length_min <= 10080;
"""
table_ref = dataset_ref.table("combined_tripdatav3")

table = client.get_table(table_ref)

client.list_rows(table, max_results=10).to_dataframe()

ride_id  rideable_type start_station_name start_station_id  \
0  C3C9652F336344D3  electric_bike               None             None   
1  AAB2B18C19B9E99C  electric_bike               None             None   
2  2D6DA95AA736089A  electric_bike               None             None   
3  8E67A84F7428B0F1  electric_bike               None             None   
4  9B72F56AAACF0971  electric_bike               None             None   
5  4D00CB831F98048F  electric_bike               None             None   
6  1D9F7400DFD86FDA  electric_bike               None             None   
7  92240F641A125443  electric_bike               None             None   
8  F9B2B0E398E6A8C6  electric_bike               None             None   
9  A681B06CAB8A4EC1  electric_bike               None             None   

  end_station_name end_station_id member_casual  start_lat  start_lng  \
0             None           None        member      42.00     -87.67   
1             None           None        casual      42.00     -87.67   
2             None           None        casual      41.99     -87.67   
3             None           None        casual      42.00     -87.70   
4             None           None        casual      41.95     -87.66   
5             None           None        member      42.00     -87.70   
6             None           None        casual      42.00     -87.70   
7             None           None        casual      42.01     -87.67   
8             None           None        casual      42.00     -87.70   
9             None           None        member      42.00     -87.71   

   end_lat  end_lng         day weekday_name month_name  hour_of_day  \
0     42.0   -87.67  2022-09-06      Tuesday  September           20   
1     42.0   -87.67  2022-08-27     Saturday     August           19   
2     42.0   -87.67  2022-12-11       Sunday   December           16   
3     42.0   -87.70  2022-05-13       Friday        May           18   
4     42.0   -87.86  2022-07-25       Monday       July           20   
5     42.0   -87.69  2022-05-13       Friday        May           19   
6     42.0   -87.70  2022-06-13       Monday       June           15   
7     42.0   -87.67  2022-05-29       Sunday        May           14   
8     42.0   -87.70  2022-07-19      Tuesday       July           20   
9     42.0   -87.70  2022-07-07     Thursday       July           23   

          ended_at_dt       started_at_dt  ride_length_min   dist_meters  \
0 2022-09-06 20:38:23 2022-09-06 20:26:11        12.200000      0.000000   
1 2022-08-27 19:10:05 2022-08-27 19:09:54         0.183333      0.000000   
2 2022-12-11 16:50:06 2022-12-11 16:46:08         3.966667   1111.951012   
3 2022-05-13 19:01:18 2022-05-13 18:56:46         4.533333      0.000000   
4 2022-07-25 22:01:41 2022-07-25 20:47:47        73.900000  17443.074299   
5 2022-05-13 20:01:14 2022-05-13 19:57:53         3.350000    826.340640   
6 2022-06-13 15:35:36 2022-06-13 15:26:50         8.766667      0.000000   
7 2022-05-29 14:50:56 2022-05-29 14:45:42         5.233333   1111.951012   
8 2022-07-19 20:13:20 2022-07-19 20:13:07         0.216667      0.000000   
9 2022-07-07 23:49:50 2022-07-07 23:42:02         7.800000    826.340640   

   dist_miles  
0    0.000000  
1    0.000000  
2    0.690934  
3    0.000000  
4   10.838624  
5    0.513464  
6    0.000000  
7    0.690934  
8    0.000000  
9    0.513464

It's suggested to check if 0.017 minutes is a valid value. For now, we assume it is correct.

In [13]:
query11="""

SELECT
  AVG(ride_length_min) AS mean_ride_length,
  MAX(ride_length_min) AS max_ride_length,
  MIN(ride_length_min) AS min_ride_length
FROM my-project-010823-374220.cyclistic_data.combined_tripdatav3;
"""
query_job = client.query(query11)

rideable_count = query_job.to_dataframe()

rideable_count.head()

mean_ride_length  max_ride_length  min_ride_length
0          17.84041     10040.383333         0.016667

<a id="10"></a>
# 4. ANALYZE


The query below pulls data by Member and Casual rides. We look at number of trips, average ride length and average distance per group. Length is measured in minutes and distance in miles.

The calculated field distance in miles is not accurate. The distance is calculated as the difference between the coordinates of the starting point and the ending point of a trip without considering for example, the extra miles riders take when going around town.

In [14]:
import warnings
warnings.filterwarnings('ignore')

query12="""

SELECT 
    member_casual,
    COUNT (DISTINCT ride_id) AS number_of_trips,
    AVG(ride_length_min) AS mean_ride_length_min,
    AVG(dist_miles) AS mean_dist_miles
FROM my-project-010823-374220.cyclistic_data.combined_tripdatav3
GROUP BY member_casual;
"""
query_job = client.query(query12)

rideable_count = query_job.to_dataframe()

rideable_count.head()

member_casual  number_of_trips  mean_ride_length_min  mean_dist_miles
0        member          3463689             12.581284         1.304863
1        casual          2364341             25.544872         1.347325

The query below is grouping the data by member_casual and rideable_type. It shows the number of trips and average ride length by bike type (electric and classic).

In [15]:
import warnings
warnings.filterwarnings('ignore')

query13="""

SELECT 
    member_casual,rideable_type,
    COUNT (DISTINCT ride_id) AS number_of_trips,
    AVG(ride_length_min) AS mean_ride_length_min
FROM my-project-010823-374220.cyclistic_data.combined_tripdatav3
GROUP BY member_casual, rideable_type
HAVING rideable_type IS NOT NULL
ORDER BY number_of_trips DESC;
"""
query_job = client.query(query13)

rideable_count = query_job.to_dataframe()

rideable_count.head()

member_casual  rideable_type  number_of_trips  mean_ride_length_min
0        member   classic_bike          1761021             13.765388
1        member  electric_bike          1702668             11.356599
2        casual  electric_bike          1280009             15.997734
3        casual   classic_bike           905769             28.561918

The query below pulls data by month for both Member and Casual rides. We're looking into number of trips and average ride length in minutes.

In [16]:
import warnings
warnings.filterwarnings('ignore')

query14="""

SELECT 
    member_casual,month_name,
    COUNT (DISTINCT ride_id) AS number_of_trips,
    AVG(ride_length_min) AS mean_ride_length_min
FROM my-project-010823-374220.cyclistic_data.combined_tripdatav3
GROUP BY member_casual,month_name
ORDER BY member_casual, number_of_trips DESC;
"""
query_job = client.query(query14)

rideable_count = query_job.to_dataframe()

rideable_count.head(24)

member_casual month_name  number_of_trips  mean_ride_length_min
0         casual       July           405948             26.699072
1         casual       June           368914             27.476252
2         casual     August           358800             25.613256
3         casual  September           296583             23.868264
4         casual        May           280352             29.009943
5         casual    October           208909             21.706275
6         casual      April           126380             27.601541
7         casual   November           100726             19.126215
8         casual      March            89854             29.736294
9         casual   December            44879             18.318618
10        casual   February            43003             19.871690
11        casual    January            39993             18.053466
12        member     August           426969             13.385419
13        member       July           417403             13.719400
14        member  September           404603             12.976986
15        member       June           400116             14.000388
16        member        May           354423             13.367431
17        member    October           349659             11.959920
18        member      April           244820             11.492967
19        member   November           236935             11.132514
20        member      March           194150             11.958945
21        member    January           150288             10.362109
22        member   February           147422             10.714801
23        member   December           136901             10.620341

The query below pulls data by day of the week for both Member and Casual rides. We're looking into number of trips and average ride length in minutes.

In [17]:
query15="""

SELECT 
    member_casual,weekday_name,
    COUNT (DISTINCT ride_id) AS number_of_trips,
    AVG(ride_length_min) AS mean_ride_length_min
FROM my-project-010823-374220.cyclistic_data.combined_tripdatav3
GROUP BY member_casual,weekday_name
ORDER BY member_casual, number_of_trips DESC;
"""
query_job = client.query(query15)

rideable_count = query_job.to_dataframe()

rideable_count.head(14)

member_casual weekday_name  number_of_trips  mean_ride_length_min
0         casual     Saturday           478296             28.299942
1         casual       Sunday           398464             29.184926
2         casual       Friday           338703             24.540990
3         casual     Thursday           313563             23.019658
4         casual       Monday           283240             25.809558
5         casual    Wednesday           279839             22.001323
6         casual      Tuesday           272236             22.901267
7         member     Thursday           547354             12.156972
8         member      Tuesday           546578             11.935146
9         member    Wednesday           542401             11.971836
10        member       Monday           488595             12.154774
11        member       Friday           481154             12.399277
12        member     Saturday           454781             14.046114
13        member       Sunday           402826             13.936131

The query below pulls data by hour of the day for both Member and Casual rides. We're looking into number of trips and average ride length in minutes.

In [18]:
query16="""

SELECT 
    member_casual,hour_of_day,
    COUNT (DISTINCT ride_id) AS number_of_trips,
    AVG(ride_length_min) AS mean_ride_length_min
FROM my-project-010823-374220.cyclistic_data.combined_tripdatav3
GROUP BY member_casual,hour_of_day
ORDER BY member_casual, hour_of_day;
"""
query_job = client.query(query16)

rideable_count = query_job.to_dataframe()

rideable_count.head(48)


member_casual  hour_of_day  number_of_trips  mean_ride_length_min
0         casual            0            46938             24.718709
1         casual            1            30363             29.497129
2         casual            2            18837             29.266814
3         casual            3            11174             31.767287
4         casual            4             7747             28.547571
5         casual            5            12792             19.655894
6         casual            6            30566             17.495425
7         casual            7            53361             16.759088
8         casual            8            71950             18.739343
9         casual            9            73913             23.720995
10        casual           10            94648             28.529406
11        casual           11           123543             28.529731
12        casual           12           146733             28.640559
13        casual           13           153322             29.207673
14        casual           14           163391             28.367407
15        casual           15           181875             27.350054
16        casual           16           202300             25.735639
17        casual           17           224200             23.790098
18        casual           18           200311             24.312620
19        casual           19           153361             24.504325
20        casual           20           113373             24.627912
21        casual           21            96998             23.820753
22        casual           22            87410             23.422692
23        casual           23            65235             24.501043
24        member            0            37201             12.701014
25        member            1            22741             13.270873
26        member            2            13262             12.764185
27        member            3             8206             13.049596
28        member            4             9138             12.887725
29        member            5            32935             10.324357
30        member            6            95282             10.890873
31        member            7           179365             11.486242
32        member            8           214252             11.357522
33        member            9           150027             11.496600
34        member           10           141047             12.455203
35        member           11           168046             12.787625
36        member           12           193668             12.386479
37        member           13           192462             12.420086
38        member           14           191277             12.858274
39        member           15           230152             12.884379
40        member           16           303246             13.128492
41        member           17           362801             13.429561
42        member           18           293281             13.161434
43        member           19           211750             12.958661
44        member           20           148923             12.731266
45        member           21           117007             12.608109
46        member           22            89116             12.532611
47        member           23            58504             12.782373

The query below is extracting the top 200 routes for Members and Casual rides. Routes were taken as unique combinations of start and end stations grouped by the number of times that route was ridden during the year. The trips taken in the top 200 routes accounted for about 7% of all total trips for Casual rides, and 5% for Member rides. 

We excluded 5,938 NULLS found for end_lat and end_lng.

In [19]:
query17="""

SELECT 
   member_casual, TRIM(start_station_name) AS new_start_station_name, 
   TRIM(start_station_id) AS new_start_station_id,
   TRIM(end_station_name) AS new_end_station_name,
   COUNT(ride_id) AS number_of_trips,
   start_lat, start_lng, end_lat, end_lng
FROM my-project-010823-374220.cyclistic_data.combined_tripdatav3
GROUP BY member_casual, new_start_station_name, start_station_id, new_end_station_name, start_lat, start_lng, end_lat, end_lng
HAVING member_casual="casual" AND new_start_station_name IS NOT NULL AND end_lat IS NOT NULL AND
   end_lng IS NOT NULL
ORDER BY number_of_trips DESC
LIMIT 500;
"""
query_job = client.query(query17)

rideable_count = query_job.to_dataframe()

rideable_count.head(10)

member_casual              new_start_station_name new_start_station_id  \
0        casual             Streeter Dr & Grand Ave                13022   
1        casual   DuSable Lake Shore Dr & Monroe St                13300   
2        casual   DuSable Lake Shore Dr & Monroe St                13300   
3        casual               Michigan Ave & Oak St                13042   
4        casual             Streeter Dr & Grand Ave                13022   
5        casual                     Montrose Harbor         TA1308000012   
6        casual  DuSable Lake Shore Dr & North Blvd               LF-005   
7        casual                     Millennium Park                13008   
8        casual                 Theater on the Lake         TA1308000001   
9        casual             Streeter Dr & Grand Ave                13022   

                 new_end_station_name  number_of_trips  start_lat  start_lng  \
0             Streeter Dr & Grand Ave             8094  41.892278 -87.612043   
1   DuSable Lake Shore Dr & Monroe St             4883  41.880958 -87.616743   
2             Streeter Dr & Grand Ave             3946  41.880958 -87.616743   
3               Michigan Ave & Oak St             2321  41.900960 -87.623777   
4   DuSable Lake Shore Dr & Monroe St             2208  41.892278 -87.612043   
5                     Montrose Harbor             2117  41.963982 -87.638181   
6  DuSable Lake Shore Dr & North Blvd             1871  41.911722 -87.626804   
7                     Millennium Park             1814  41.881032 -87.624084   
8                 Theater on the Lake             1768  41.926277 -87.630834   
9  DuSable Lake Shore Dr & North Blvd             1751  41.892278 -87.612043   

     end_lat    end_lng  
0  41.892278 -87.612043  
1  41.880958 -87.616743  
2  41.892278 -87.612043  
3  41.900960 -87.623777  
4  41.880958 -87.616743  
5  41.963982 -87.638181  
6  41.911722 -87.626804  
7  41.881032 -87.624084  
8  41.926277 -87.630834  
9  41.911722 -87.626804

In [20]:
query18="""

SELECT 
   member_casual, TRIM(start_station_name) AS new_start_station_name, 
   TRIM(start_station_id) AS new_start_station_id,
   TRIM(end_station_name) AS new_end_station_name,
   COUNT(ride_id) AS number_of_trips,
   start_lat, start_lng, end_lat, end_lng
FROM my-project-010823-374220.cyclistic_data.combined_tripdatav3
GROUP BY member_casual, new_start_station_name, start_station_id, new_end_station_name, start_lat, start_lng, end_lat, end_lng
HAVING member_casual="member" AND new_start_station_name IS NOT NULL AND end_lat IS NOT NULL AND
   end_lng IS NOT NULL
ORDER BY number_of_trips DESC
LIMIT 500;
"""
query_job = client.query(query18)

rideable_count = query_job.to_dataframe()

rideable_count.head(10)

member_casual    new_start_station_name new_start_station_id  \
0        member       Ellis Ave & 60th St         KA1503000014   
1        member  University Ave & 57th St         KA1503000071   
2        member       Ellis Ave & 60th St         KA1503000014   
3        member       Ellis Ave & 55th St         KA1504000076   
4        member        State St & 33rd St                13216   
5        member     Calumet Ave & 33rd St                13217   
6        member  University Ave & 57th St         KA1503000071   
7        member       Ellis Ave & 60th St         KA1503000014   
8        member     Kimbark Ave & 53rd St         TA1309000037   
9        member       Ellis Ave & 58th St         TA1309000011   

       new_end_station_name  number_of_trips  start_lat  start_lng    end_lat  \
0  University Ave & 57th St             5546  41.785097 -87.601073  41.791478   
1       Ellis Ave & 60th St             5202  41.791478 -87.599861  41.785097   
2       Ellis Ave & 55th St             4799  41.785097 -87.601073  41.794301   
3       Ellis Ave & 60th St             4518  41.794301 -87.601450  41.785097   
4     Calumet Ave & 33rd St             2415  41.834734 -87.625813  41.834900   
5        State St & 33rd St             2312  41.834900 -87.617930  41.834734   
6     Kimbark Ave & 53rd St             2006  41.791478 -87.599861  41.799568   
7       Ellis Ave & 58th St             1881  41.785097 -87.601073  41.788746   
8  University Ave & 57th St             1796  41.799568 -87.594747  41.791478   
9       Ellis Ave & 60th St             1656  41.788746 -87.601334  41.785097   

     end_lng  
0 -87.599861  
1 -87.601073  
2 -87.601450  
3 -87.601073  
4 -87.617930  
5 -87.625813  
6 -87.594747  
7 -87.601334  
8 -87.599861  
9 -87.601073

<a id="11"></a>
# 5. SHARE

As a reminder, customers who purchase single-ride or full-day passes are referred to as Casual riders. Customers who purchase annual memberships are Cyclistic Members.

59.3% of total rides were Member trips, and 40.6% were Casual rides. We don't know exactly what percentage of total customers Members and Casuals are. However, we know that
in the year from March 2022 through February 2023, Members made the most trips, 59.3% of
the total.

The average ride length in minutes was 12.58 for Members and 25.54 for Casuals. This seems to indicate that Members usually ride bikes to commute while Casuals ride mostly for tourism and leisure, taking more time per ride.

The calculated field distance in miles is not accurate. The distance is calculated as the difference between the coordinates of the starting point and the ending point of a trip. For Members who commute this number is more valid, but for Casuals is not. Casuals might start in one station and go all around town before ending their trip at an ending station. The distance for Casuals is the distance between starting and ending points without considering for example, the extra miles they took when going around town.

In [21]:
%%HTML
<div class='tableauPlaceholder' id='viz1680125504300' style='position: relative'><noscript><a href='#'><img alt='Dashboard 1 ' src='https:&#47;&#47;public.tableau.com&#47;static&#47;images&#47;Cy&#47;CyclisticCaseStudyViz1&#47;Dashboard1&#47;1_rss.png' style='border: none' /></a></noscript><object class='tableauViz'  style='display:none;'><param name='host_url' value='https%3A%2F%2Fpublic.tableau.com%2F' /> <param name='embed_code_version' value='3' /> <param name='site_root' value='' /><param name='name' value='CyclisticCaseStudyViz1&#47;Dashboard1' /><param name='tabs' value='no' /><param name='toolbar' value='yes' /><param name='static_image' value='https:&#47;&#47;public.tableau.com&#47;static&#47;images&#47;Cy&#47;CyclisticCaseStudyViz1&#47;Dashboard1&#47;1.png' /> <param name='animate_transition' value='yes' /><param name='display_static_image' value='yes' /><param name='display_spinner' value='yes' /><param name='display_overlay' value='yes' /><param name='display_count' value='yes' /><param name='language' value='en-US' /></object></div> <script type='text/javascript'> var divElement = document.getElementById('viz1680125504300'); var vizElement = divElement.getElementsByTagName('object')[0]; if ( divElement.offsetWidth > 800 ) { vizElement.style.width='1000px';vizElement.style.height='827px';} else if ( divElement.offsetWidth > 500 ) { vizElement.style.width='1000px';vizElement.style.height='827px';} else { vizElement.style.width='100%';vizElement.style.height='1377px';} var scriptElement = document.createElement('script'); scriptElement.src = 'https://public.tableau.com/javascripts/api/viz_v1.js'; vizElement.parentNode.insertBefore(scriptElement, vizElement); </script>

Casual rides showed a preference for electric bikes: 58.6% of all rides used electric bikes while 41.4% used classic bikes. Casual riders seem to prefer electric bikes because these are faster, more reliable for longer trips and easier to use.

Members on the other hand used both electric and classic bikes almost at the same rate, 50.8% of rides used electric bikes while 49% used classic bikes.

In average length per ride, classic bikes take longer per ride for both Members and Casuals. For Members that difference is small (13.7 minutes for classic vs. 11.3 for electric bikes). On the other hand, Casual rides with electric bikes took much less time per ride than classic bikes (16 vs. 28.5 minutes); we can assume that electric bikes were a much faster way of transportation than classic bikes.

In [22]:
%%HTML
<div class='tableauPlaceholder' id='viz1680134810481' style='position: relative'><noscript><a href='#'><img alt='Dashboard 1 ' src='https:&#47;&#47;public.tableau.com&#47;static&#47;images&#47;Cy&#47;CyclisticDataRideableTypeViz5&#47;Dashboard1&#47;1_rss.png' style='border: none' /></a></noscript><object class='tableauViz'  style='display:none;'><param name='host_url' value='https%3A%2F%2Fpublic.tableau.com%2F' /> <param name='embed_code_version' value='3' /> <param name='site_root' value='' /><param name='name' value='CyclisticDataRideableTypeViz5&#47;Dashboard1' /><param name='tabs' value='no' /><param name='toolbar' value='yes' /><param name='static_image' value='https:&#47;&#47;public.tableau.com&#47;static&#47;images&#47;Cy&#47;CyclisticDataRideableTypeViz5&#47;Dashboard1&#47;1.png' /> <param name='animate_transition' value='yes' /><param name='display_static_image' value='yes' /><param name='display_spinner' value='yes' /><param name='display_overlay' value='yes' /><param name='display_count' value='yes' /><param name='language' value='en-US' /></object></div> <script type='text/javascript'> var divElement = document.getElementById('viz1680134810481'); var vizElement = divElement.getElementsByTagName('object')[0]; if ( divElement.offsetWidth > 800 ) { vizElement.style.width='1000px';vizElement.style.height='827px';} else if ( divElement.offsetWidth > 500 ) { vizElement.style.width='1000px';vizElement.style.height='827px';} else { vizElement.style.width='100%';vizElement.style.height='727px';} var scriptElement = document.createElement('script');scriptElement.src = 'https://public.tableau.com/javascripts/api/viz_v1.js';vizElement.parentNode.insertBefore(scriptElement, vizElement);</script>

For Casuals, the number of rides peak during the summer months of June, July and August. Casual rides seem to be closely related to the weather, there is also activity between the spring months of April and May, and the fall months of September and October. During the colder months of winter (December, January and February) Casual rides declined sharply to less than 50,000 rides a month.

Member rides tend to be more consistent throughout the year with the highest months being July and August. There is a similar pattern to that of Casual rides, however the colder months in winter show much more activity with about 140,000 rides per month.

In average time per ride, Casual rides peak in the month of March with 30 minutes per ride. The months of March through August have over 25 minutes of average ride duration. For Member rides the average length is more consistent throughout the year fluctuating between 10 and 14 minutes. This pattern seems closer to a commuter’s time slightly peaking during the summer months.

In [23]:
%%HTML
<div class='tableauPlaceholder' id='viz1680185250343' style='position: relative'><noscript><a href='#'><img alt='Dashboard 1 ' src='https:&#47;&#47;public.tableau.com&#47;static&#47;images&#47;Nu&#47;Number_of_Trips_by&#47;Dashboard1&#47;1_rss.png' style='border: none' /></a></noscript><object class='tableauViz'  style='display:none;'><param name='host_url' value='https%3A%2F%2Fpublic.tableau.com%2F' /> <param name='embed_code_version' value='3' /> <param name='site_root' value='' /><param name='name' value='Number_of_Trips_by&#47;Dashboard1' /><param name='tabs' value='no' /><param name='toolbar' value='yes' /><param name='static_image' value='https:&#47;&#47;public.tableau.com&#47;static&#47;images&#47;Nu&#47;Number_of_Trips_by&#47;Dashboard1&#47;1.png' /> <param name='animate_transition' value='yes' /><param name='display_static_image' value='yes' /><param name='display_spinner' value='yes' /><param name='display_overlay' value='yes' /><param name='display_count' value='yes' /><param name='language' value='en-US' /></object></div> <script type='text/javascript'> var divElement = document.getElementById('viz1680185250343'); var vizElement = divElement.getElementsByTagName('object')[0]; if ( divElement.offsetWidth > 800 ) { vizElement.style.width='800px';vizElement.style.height='827px';} else if ( divElement.offsetWidth > 500 ) { vizElement.style.width='800px';vizElement.style.height='827px';} else { vizElement.style.width='100%';vizElement.style.height='727px';} var scriptElement = document.createElement('script'); scriptElement.src = 'https://public.tableau.com/javascripts/api/viz_v1.js'; vizElement.parentNode.insertBefore(scriptElement, vizElement); </script>

Casual rides peak on Saturdays and Sundays at over 400,000 rides. During the week from Monday through Friday the number of rides stay at around 300,000 rides. For Members the opposite happens, Saturday and Sunday are the days with the lowest number of rides, while from Monday through Friday this number stays around 500,000 rides.

Average length per ride follows the same pattern as number of trips. During the weekend casuals take longer trips between 28 and 29 minutes. Member rides are more consistent at between 12 to 14 minutes during the entire week.

In [24]:
%%HTML
<div class='tableauPlaceholder' id='viz1680186042156' style='position: relative'><noscript><a href='#'><img alt='Dashboard 1 ' src='https:&#47;&#47;public.tableau.com&#47;static&#47;images&#47;Cy&#47;CyclisticDatabyDayofWeek&#47;Dashboard1&#47;1_rss.png' style='border: none' /></a></noscript><object class='tableauViz'  style='display:none;'><param name='host_url' value='https%3A%2F%2Fpublic.tableau.com%2F' /> <param name='embed_code_version' value='3' /> <param name='site_root' value='' /><param name='name' value='CyclisticDatabyDayofWeek&#47;Dashboard1' /><param name='tabs' value='no' /><param name='toolbar' value='yes' /><param name='static_image' value='https:&#47;&#47;public.tableau.com&#47;static&#47;images&#47;Cy&#47;CyclisticDatabyDayofWeek&#47;Dashboard1&#47;1.png' /> <param name='animate_transition' value='yes' /><param name='display_static_image' value='yes' /><param name='display_spinner' value='yes' /><param name='display_overlay' value='yes' /><param name='display_count' value='yes' /><param name='language' value='en-US' /></object></div> <script type='text/javascript'> var divElement = document.getElementById('viz1680186042156'); var vizElement = divElement.getElementsByTagName('object')[0]; if ( divElement.offsetWidth > 800 ) { vizElement.style.width='800px';vizElement.style.height='827px';} else if ( divElement.offsetWidth > 500 ) { vizElement.style.width='800px';vizElement.style.height='827px';} else { vizElement.style.width='100%';vizElement.style.height='727px';} var scriptElement = document.createElement('script'); scriptElement.src = 'https://public.tableau.com/javascripts/api/viz_v1.js'; vizElement.parentNode.insertBefore(scriptElement, vizElement); </script>

Most rides for both groups (Members and Casuals) take place in the afternoon between the hours of 4pm and 6pm. The time between 12pm and 3pm have the second highest number of trips per hour, followed by 7pm. 

Casuals prefer the afternoons with most trips happening between 12pm and 7pm. 

Members follow a working schedule with most trips happening during the commuting time of 4pm to 6pm. Although the morning hours have less activity, the commuting hours between 6am and 8am have about three times as many trips as Casual rides.

For the average time per ride, Member rides are more consistent with 10 to 13 minutes throughout the day.  For Casuals, late morning and early afternoon rides (between 10am and 3pm) tend to be longer at 29 to 27 minutes.  For Casuals, the average length ride peaks at 32 minutes at 3am. This is due to the presence of 7 outliers that report trips for more than 5,000 hours at the hour of 3am. We would need to investigate that to validate this data. Although Casual rides in the early morning of the day (1am to 4am) seem a little high, we can observe that Casuals prefer late morning and afternoon rides at an average of about 29 minutes.

In [25]:
%%HTML
<div class='tableauPlaceholder' id='viz1680186843215' style='position: relative'><noscript><a href='#'><img alt='Dashboard 1 ' src='https:&#47;&#47;public.tableau.com&#47;static&#47;images&#47;Cy&#47;Cyclistic_16795966693180&#47;Dashboard1&#47;1_rss.png' style='border: none' /></a></noscript><object class='tableauViz'  style='display:none;'><param name='host_url' value='https%3A%2F%2Fpublic.tableau.com%2F' /> <param name='embed_code_version' value='3' /> <param name='site_root' value='' /><param name='name' value='Cyclistic_16795966693180&#47;Dashboard1' /><param name='tabs' value='no' /><param name='toolbar' value='yes' /><param name='static_image' value='https:&#47;&#47;public.tableau.com&#47;static&#47;images&#47;Cy&#47;Cyclistic_16795966693180&#47;Dashboard1&#47;1.png' /> <param name='animate_transition' value='yes' /><param name='display_static_image' value='yes' /><param name='display_spinner' value='yes' /><param name='display_overlay' value='yes' /><param name='display_count' value='yes' /><param name='language' value='en-US' /></object></div> <script type='text/javascript'> var divElement = document.getElementById('viz1680186843215'); var vizElement = divElement.getElementsByTagName('object')[0]; if ( divElement.offsetWidth > 800 ) { vizElement.style.width='800px';vizElement.style.height='827px';} else if ( divElement.offsetWidth > 500 ) { vizElement.style.width='800px';vizElement.style.height='827px';} else { vizElement.style.width='100%';vizElement.style.height='727px';} var scriptElement = document.createElement('script'); scriptElement.src = 'https://public.tableau.com/javascripts/api/viz_v1.js'; vizElement.parentNode.insertBefore(scriptElement, vizElement); </script>

The top 200 routes for rides, were taken as unique combinations of start and end stations grouped by the number of times that route was ridden during the year. The trips taken in the top 200 routes accounted for about 7% of all total trips for Casual rides, and 5% for Member rides. 

Casual routes are concentrated in the downtown area of Chicago, with the station at Streeter Dr. and Grand Ave. being the start station that originated the most routes. This shows that most Casual rides are taken by tourists and visitors looking to get around downtown to visit local attractions.

On the other hand, Member trips were taken across a wider section of the map, reaching farther areas from the center of the city. In fact, the top Member route was taken from the start station of University Ave and 57th St. which is located at Hyde Park. This makes sense because most of Members are commuters.

In [26]:
%%HTML
<div class='tableauPlaceholder' id='viz1680207154138' style='position: relative'><noscript><a href='#'><img alt='Dashboard 1 ' src='https:&#47;&#47;public.tableau.com&#47;static&#47;images&#47;To&#47;Top500CasualRides&#47;Dashboard1&#47;1_rss.png' style='border: none' /></a></noscript><object class='tableauViz'  style='display:none;'><param name='host_url' value='https%3A%2F%2Fpublic.tableau.com%2F' /> <param name='embed_code_version' value='3' /> <param name='site_root' value='' /><param name='name' value='Top500CasualRides&#47;Dashboard1' /><param name='tabs' value='no' /><param name='toolbar' value='yes' /><param name='static_image' value='https:&#47;&#47;public.tableau.com&#47;static&#47;images&#47;To&#47;Top500CasualRides&#47;Dashboard1&#47;1.png' /> <param name='animate_transition' value='yes' /><param name='display_static_image' value='yes' /><param name='display_spinner' value='yes' /><param name='display_overlay' value='yes' /><param name='display_count' value='yes' /><param name='language' value='en-US' /></object></div> <script type='text/javascript'> var divElement = document.getElementById('viz1680207154138'); var vizElement = divElement.getElementsByTagName('object')[0]; vizElement.style.width='800px';vizElement.style.height='827px'; var scriptElement = document.createElement('script'); scriptElement.src = 'https://public.tableau.com/javascripts/api/viz_v1.js'; vizElement.parentNode.insertBefore(scriptElement, vizElement); </script>

In [27]:
%%HTML
<div class='tableauPlaceholder' id='viz1680195290954' style='position: relative'><noscript><a href='#'><img alt='Dashboard 1 ' src='https:&#47;&#47;public.tableau.com&#47;static&#47;images&#47;WY&#47;WY5M526F6&#47;1_rss.png' style='border: none' /></a></noscript><object class='tableauViz'  style='display:none;'><param name='host_url' value='https%3A%2F%2Fpublic.tableau.com%2F' /> <param name='embed_code_version' value='3' /> <param name='path' value='shared&#47;WY5M526F6' /> <param name='toolbar' value='yes' /><param name='static_image' value='https:&#47;&#47;public.tableau.com&#47;static&#47;images&#47;WY&#47;WY5M526F6&#47;1.png' /> <param name='animate_transition' value='yes' /><param name='display_static_image' value='yes' /><param name='display_spinner' value='yes' /><param name='display_overlay' value='yes' /><param name='display_count' value='yes' /><param name='language' value='en-US' /></object></div> <script type='text/javascript'> var divElement = document.getElementById('viz1680195290954'); var vizElement = divElement.getElementsByTagName('object')[0]; if ( divElement.offsetWidth > 800 ) { vizElement.style.width='1000px';vizElement.style.height='827px';} else if ( divElement.offsetWidth > 500 ) { vizElement.style.width='1000px';vizElement.style.height='827px';} else { vizElement.style.width='100%';vizElement.style.height='727px';} var scriptElement = document.createElement('script');  scriptElement.src = 'https://public.tableau.com/javascripts/api/viz_v1.js'; vizElement.parentNode.insertBefore(scriptElement, vizElement); </script>

<a id="12"></a>
# 6. ACT


<a id="13"></a>
### A. CONCLUSIONS

* We concluded that Members are mostly commuters while Casual riders are tourists and visitors that use bikes to get to local attractions and events.

* 59.3% of total rides were Member trips, 40.6% were Casual rides.

* The average ride length in minutes was 12.58 for Members and 25.54 for Casuals. This seems to indicate that Members usually ride bikes to commute while Casuals ride mostly for tourism and leisure, taking more time per ride.

* Casual rides showed a preference for electric bikes, probably because these are faster and more reliable for longer trips. The Casual average ride length for electric bikes was much lower than that for classic bikes (16 vs. 28.6 minutes). 

* Members on the other hand used both electric and classic bikes at the same rate, and at a consistent average length per ride (11 to 13 minutes). 

* For Casuals, the number of rides peak during the summer months of June, July and August. Casual rides seem to be closely related to the weather; during the colder months of winter (December, January and February) Casual rides declined sharply.

* Members number of rides and average length per ride (10 to 14 minutes) tend to be more consistent throughout the months of the year.

* Casual rides peak on Saturdays and Sundays. For Members the opposite happens, Saturday and Sunday are the days with the lowest number of rides. 

* During the weekend Casuals take longer trips at 28 to 29 minutes per ride. Member rides are more consistent at 12 to 14 minutes for all days of the week.

* Members follow a working schedule with most trips happening during the commuting time of 4pm to 6pm. Casuals prefer the afternoons with most trips happening between 12pm and 7pm.

* For the average time per ride, Member rides are more consistent with 10 to 13 minutes throughout the hours of the day.  For Casuals, the average length per ride tends to be longer, almost twice as much as Member rides.  

* Casual routes are concentrated in the downtown area of Chicago, showing that most Casual rides are taken by tourists and visitors looking to get around downtown to visit local attractions.

* On the other hand, Member trips were more of the commuter type. Rides were taken across a wider section of the map, reaching farther areas from the center of the city. 

<a id="14"></a>
### B. RECOMMENDATIONS

We concluded that Members and Casual riders are two different customer segments that have different needs. Converting some Casual riders into Members might be possible if there is a group of Casual riders that use the bikes to commute but don’t want to commit to an annual membership. It would be recommendable to have a marketing strategy that communicates to these Casual riders that they can save money with a membership. The next step though would be to have a customer survey to find out how often this is happening.

However, the group from Casual riders that acquire a membership will probably not be a massive move. We could estimate how big this group is by the results of the customer survey and then analyze financial information such as cost and profits. This way we could better measure the financial impact of converting some Casual riders to annual Members. 

We believe that keeping pricing flexibility is very important. Communicating the benefits of each pricing plan is critical as well. Casual riders can be engaged by digital marketing through social media. Also, campaigns such as Google Display would be useful to advertise in thousands of websites with touristic content. 